addons/sale_project/models/project.py

```diff
@@ -45,6 +47,27 @@ def _compute_has_any_so_to_invoice(self):
                     has_any_so_to_invoice = has_any_so_to_invoice or (task.sale_order_id and task.sale_order_id.invoice_status == "to invoice")
                 project.has_any_so_to_invoice = has_any_so_to_invoice
 
+    @api.depends('sale_order_id')
+    def _compute_display_create_invoice_buttons(self):
+        primary, secondary = False, False
+        if not self.user_has_groups('sales_team.group_sale_salesman_all_leads'):
+            self.update({
+                'display_create_invoice_primary': primary,
+                'display_create_invoice_secondary': secondary,
+            })
+        else:
+            for project in self:
+                if project.has_any_so_to_invoice:
```

Comment:
The compute of `has_any_so_to_invoice` field checks in the SO of the project but also in all tasks related to this project, so I think the `else` statement in this method will never be satisfied.
What do you think ? 🤔

Reply:
I'm not sure what you mean: if a project or its tasks do not have a sale order id whose status is "to invoice", then we should enter in the else statement, right?